cartography/intel/aws/ec2.py

```diff
@@ -606,3 +612,259 @@ def sync_load_balancers(session, boto3_session, regions, current_aws_account_id,
+    # paginator not supported by boto
+    return client.describe_vpcs()
+
+
+def load_ec2_vpcs(session, data, current_aws_account_id, aws_update_tag):
+    # https://github.com/lyft/cartography/graphs/traffic
+    # {
+    #     "Vpcs": [
+    #         {
+    #             "VpcId": "vpc-a01106c2",
+    #             "InstanceTenancy": "default",
+    #             "Tags": [
+    #                 {
+    #                     "Value": "MyVPC",
+    #                     "Key": "Name"
+    #                 }
+    #             ],
+    #             "CidrBlockAssociations": [
+    #                 {
+    #                     "AssociationId": "vpc-cidr-assoc-a26a41ca",
+    #                     "CidrBlock": "10.0.0.0/16",
+    #                     "CidrBlockState": {
+    #                         "State": "associated"
+    #                     }
+    #                 }
+    #             ],
+    #             "State": "available",
+    #             "DhcpOptionsId": "dopt-7a8b9c2d",
+    #             "CidrBlock": "10.0.0.0/16",
+    #             "IsDefault": false
+    #         }
+    #     ]
+    # }
+
+    ingest_vpc = """
+    MERGE (new_vpc:AWSVpc{id: {VpcId}})
+    ON CREATE SET new_vpc.firstseen = timestamp(), new_vpc.vpcid ={VpcId}
+    SET new_vpc.instance_tenancy = {InstanceTenancy},
+    new_vpc.state = {State},
+    new_vpc.is_default = {IsDefault},
+    new_vpc.primary_cidr_block = {PrimaryCIDRBlock},
+    new_vpc.dhcp_options_id = {DhcpOptionsId},
+    new_vpc.lastupdated = {aws_update_tag}
+    WITH new_vpc
+    MATCH (awsAccount:AWSAccount{id: {AWS_ACCOUNT_ID}})
+    MERGE (awsAccount)-[r:RESOURCE]->(new_vpc)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag}"""
+
+    for vpc in data['Vpcs']:
+        vpc_id = vpc["VpcId"]  # fail if not present
+
+        session.run(
+            ingest_vpc,
+            VpcId=vpc_id,
+            InstanceTenancy=vpc.get("InstanceTenancy", ""),
+            State=vpc.get("State", ""),
+            IsDefault=vpc.get("IsDefault", ""),
+            PrimaryCIDRBlock=vpc.get("CidrBlock", ""),
+            DhcpOptionsId=vpc.get("DhcpOptionsId", ""),
+            AWS_ACCOUNT_ID=current_aws_account_id,
+            aws_update_tag=aws_update_tag)
+
+        ingest_cidr_association_set(session,
+                                    vpc_id=vpc_id,
+                                    type="ipv4",
+                                    data=vpc.get("CidrBlockAssociationSet", []),
+                                    aws_update_tag=aws_update_tag)
+
+        ingest_cidr_association_set(session,
+                                    vpc_id=vpc_id,
+                                    type="ipv6",
+                                    data=vpc.get("Ipv6CidrBlockAssociationSet", []),
+                                    aws_update_tag=aws_update_tag)
+
+
+def ingest_cidr_association_set(session, vpc_id, type, data, aws_update_tag):
```

Comment:
fine either way. This trick was used elsewhere in EC2 ingestion (load_ec2_security_group_rule). Also avoids having 2 code block that needs to be maintained over time.

Reply:
for UNWIND, I use to have that concern but seems neo4j address the past issue. We use UNWIND extensively in our internal Ubuntu CVE mapper without problems.

Reply:
I like Alex's feedback to avoid string replacement like this. IMO it makes the code much harder to reason about. That being said, I also agree that a single query is preferable as it will help us avoid being bitten in future when maintaining this code. I think we can achieve this as follows:

1. Factor out data transformation into it's own function. Make the T in ETL explicit.
2. Pass the output of the transformation to this ingest function (should be renamed load_ to be consistent).
3. To solve the label problem, run a subsequent query that labels nodes based on an attribute they have set.

Can discuss this in person if the above is not sufficiently clear.

Reply:
Took feedback and made some changes. 
From evan, I did 1,2 but #3 I felt was adding complexity

Why not #3 
* added complexity where we needed to store a flag on the node and later on scan all new nodes for that flag and set the right type. This would have required 2 pass (ipv6 and ipv4)
* used graph to store a flag that is not needed over time. I felt we shouldn't use the graph in that way